### PR TITLE
Limit toasts' width

### LINKED ISSUE #57472: New in-editor error popups can extend off-screen
### Godot version

v4.0.alpha.custom_build [78e3e65e7]

### System information

Windows 10

### Issue description

The new error popups don't wrap text or otherwise handle longer text, so they can extend off-screen:

image

Since the errors are left-justified as a block, long errors can push shorter ones off-screen:

image

Popups don't seem to have been introduced  in v4.0.alpha1.official [31a7ddbf8], so I haven't been able to reproduce the behaviour there.

**Edit: ** The bug also appears in alpha 1.

### Steps to reproduce

Cause a sufficiently long error message to popup in the editor, e.g. with
```
@tool
extends Node

func _process(delta):
    var a
    a.loooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong_function_name()
```
attached to a node in the scene tree.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by groud**
Fixes https://github.com/godotengine/godot/issues/57472

Fixes a small visibility bug too.

### CODE REVIEW

File: editor/editor_toaster.cpp [hunk 1]
```diff
@@ -441,14 +449,31 @@ void EditorToaster::_popup_str(String p_message, Severity p_severity, String p_t
 		main_button->queue_redraw();
 	}
 
-	// Retrieve the label back then update the text.
-	Label *label = Object::cast_to<Label>(control->get_child(0)->get_child(0));
+	// Retrieve the label back, then update the text.
+	Label *label = Object::cast_to<Label>(control->get_child(0)->get_child(0)->get_child(0));
```
Comment (akien-mga) [inline editor/editor_toaster.cpp:453]: This is really hacky. Would be better to keep a pointer as member variable, like we do everywhere else.

File: editor/editor_toaster.cpp [hunk 2]
```diff
@@ -441,14 +449,31 @@ void EditorToaster::_popup_str(String p_message, Severity p_severity, String p_t
...
+
+	Size2i size = label->get_combined_minimum_size();
+	int limit_width = get_viewport_rect().size.x / 2; // Limit label size to half the viewport size.
+	if (size.x > limit_width) {
+		label->set_text_overrun_behavior(TextServer::OVERRUN_TRIM_ELLIPSIS);
+		label->set_custom_minimum_size(Size2(limit_width, 0));
+	}
+
+	// Retrieve the count label back, then update the text.
+	Label *count_label = Object::cast_to<Label>(control->get_child(0)->get_child(0)->get_child(1));
```
Comment (akien-mga) [inline editor/editor_toaster.cpp:467]: Same here.

File: editor/editor_toaster.cpp (hunk 1 above)
Comment (groud) [reply editor/editor_toaster.cpp:453]: That would mean I need to add both to the Toast struct.
I cannot simply keep the reference to the `label` / `count_label` variable before, as I need to find it again in the toast list when the goal is only to increase the count.

But yeah, it's not a big deal, I can implement it that way too.

Comment (akien-mga) [reply editor/editor_toaster.cpp:453]: Could it use a node unique name at least? This kind of code is asking for breakage and not a very good example of best UI practices :)

**Review by Calinou — APPROVED** [review]

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**